libcxx/include/__config

```diff
@@ -559,6 +559,18 @@ typedef __char32_t char32_t;
 #  endif
 
 // clang-format off
+#  define _LIBCPP_BEGIN_HIDE_FROM_ABI                                                                                  \
```

Comment:
I feel that the name of this macro is misleading, since this doesn't do the same thing as `_LIBCPP_HIDE_FROM_ABI`. Maybe just don't introduce this macro and use `_LIBCPP_BEGIN_ABI_SENSITIVE` directly?

In fact, we could also use something like `_LIBCPP_BEGIN_EXPLICIT_ABI` or `_LIBCPP_BEGIN_EXPLICIT_ABI_ANNOTATIONS` instead of `ABI_SENSITIVE`. This could also remove potential confusion about things like `join_view` not being "ABI sensitive", which they totally are (e.g. you still can't reorder data members after introducing the mechanism in this patch).